scripts/bootstrap/bootstrap-haskell.ps1

```diff
@@ -37,8 +37,8 @@ param (
     [string]$ExistingMsys2Dir,
     # Specify the cabal root directory (default: '$InstallDir\cabal')
     [string]$CabalDir,
-    # Whether to disable use of curl.exe
-    [switch]$DisableCurl,
+    # Whether to use curl.exe (default: false)
+    [switch]$UseCurl,

```

Comment:
I considered that, but the [documentation](
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-7.4#switch-parameter-design-considerations) suggested otherwise

we can ignore the suggestion and keep the $DisableCurl flag as it is, and change its value by either
* prompting the user (in interactive mode)
* always make it true ( if the user has not explicitly specified its value)

Reply:
Perhaps even "always make it true " would be a breaking change in certain cases. Therefore "prompting the user" seems to be the only good option

Reply:
What mean is that these can be used as **positional parameters**.

Changing the meaning of a flag at a certain position breaks scripts that execute the powershell script with a positional argument list.

E.g. see

```
Set-ExecutionPolicy Bypass -Scope Process -Force;[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; try { Invoke-Command -ScriptBlock ([ScriptBlock]::Create((Invoke-WebRequest https://www.haskell.org/ghcup/sh/bootstrap-haskell.ps1 -UseBasicParsing))) -ArgumentList $true } catch { Write-Error $_ }
```

The `ArgumentList` part.

Reply:
Yeah I understood that. So my idea is to keep the $DisableCurl flag as it is.

But if we want to disable curl by default, we have two choices

1. we can check if the flag has been specified; by [checking this](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-7.4#psboundparameters), and if user has not specified the param as either true or false, then we can make it true.
 
2. In the interactive mode we can prompt the user if they would like to enable curl or not.
    The default choice will be "DisableCurl = true".

   The behaviour of this script in non-interactive mode would remain same (ie DisableCurl will be false)

I think 2nd option is better.

Reply:
I don't think we should add another prompt. The only thing that needs changing is the GHCup website to include `true` for the positional parameter that disables curl.

https://github.com/haskell/ghcup-hs/blob/7f215b0891728625ba156c4291c1acba4cef7a43/docs/index.md?plain=1#L44

Reply:
Oh right, I didnt think of that. will make a PR soon.